Upgrades hammer.js and waveSurferJs

NishealJ opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
This PR Upgrades **waveSurferJs & hammer.js**
To see the status of the library upgrade please visit [here](https://docs.google.com/spreadsheets/d/17gVjhKyKBqpWEhvZ92V0AB5ruoYvy2X0Te5mrZClxWk/edit#gid=0)

For testing doc please visit [here](https://docs.google.com/document/d/1Jld0h1QGnmjfnELu6dH3V5UXIgCpSOTE-knbv_1dBOw/edit) 


## Checklist
- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

Commit:
initial upgrade

NishealJ committed:
Fix mistakes

NishealJ committed:
Fix mistakes 2

NishealJ committed:
resolve backend tests

NishealJ committed:
Removed wavesurf

NishealJ committed:
Merge remote-tracking branch 'origin/develop' into guppy_up

NishealJ commented:
Hi @apb7,
upgrading select-2 failed the testing seems like they have issues with their latest version.
I've mentioned the issues we are facing after the upgrade [here](https://docs.google.com/document/d/1hcl1sVw_ehtl5GuX7bgMPIrio4hp2c8eL12PSglI3zw/edit)  PTAL Thanks!

apb7 commented:
@NishealJ, left a comment on the doc. Thanks!

NishealJ commented:
> @NishealJ, left a comment on the doc. Thanks!

@apb7 Addressed Thanks!

NishealJ commented:
Hi @apb7 
as discussed regarding the `wavesurf.js` PTAL the tests failing [here](https://circleci.com/gh/oppia/oppia/9375?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) the [wavesurf.js](https://github.com/katspaugh/wavesurfer.js/blob/master/src/wavesurfer.js) file starts with the unexpected import syntax to the files which are not present in the third_party generated hence, this is probably diffcult to combine with other libs I'm working on this need your comments Thanks !

NishealJ committed:
wavesurfer to 1.4.0

NishealJ committed:
Merge remote-tracking branch 'origin/develop' into guppy_up

apb7 commented:
Hi @NishealJ, please upgrade this PR as discussed in the meet. Thanks!

NishealJ committed:
Revert Select 2

NishealJ committed:
Revert package-lock

NishealJ commented:
> Hi @NishealJ, please upgrade this PR as discussed in the meet. Thanks!

Hi @apb7, I've done the needed changes as discussed & [here](https://docs.google.com/document/d/1Jld0h1QGnmjfnELu6dH3V5UXIgCpSOTE-knbv_1dBOw/edit?usp=sharing) is the testing doc

NishealJ committed:
Merge remote-tracking branch 'origin/develop' into guppy_up

NishealJ commented:
Hi @seanlip & @apb7 PTAL Thanks!

seanlip commented:
Backend tests are failing.

NishealJ commented:
> Backend tests are failing.

Sorry @seanlip, this happened after I asked for a review and updated with develop.
I'll try to debug this Thanks!

NishealJ committed:
Merge remote-tracking branch 'origin/develop' into guppy_up

NishealJ commented:
Hi @seanlip & @apb7,
Backed tests are now resolved PTAL! Thanks!

seanlip reviewed (approved):
LGTM from me. @DubeySandeep could you PTAL?

DubeySandeep commented on manifest.json:
```diff
@@ -540,14 +540,14 @@
+        "rootDirPrefix": "wavesurfer.js-",
         "targetDirPrefix": "wave-surfer-js-",
-        "files": ["wavesurfer.min.js"],
+        "files": ["src/wavesurfer.js"],
```
Should we use the `dist/` files? (have you tested this? I'm not sure whether `src/wavesurfer.js` needs other dependencies from `src/` dir.)

Also, can we use the `dist/wavesurfer.min.js` (at the end we are going to minify the file)

NishealJ replied:
Hi @DubeySandeep, I agree with your point we should use `dict/` I'll add and test with the CDN provided.

DubeySandeep commented on manifest.json:
```diff
@@ -540,14 +540,14 @@
-        "rootDirPrefix": "waveSurfer-js-",
+        "version": "1.4.0",
+        "downloadFormat": "zip",
+        "url": "https://github.com/katspaugh/wavesurfer.js/archive/1.4.0.zip",
```
1.4.0 is the latest stable version?

DubeySandeep replied:
btw, if we need just one file do we have to download the entire repo? (isn't there any way to download the dist/ files only?)

DubeySandeep replied:
I found a cdn here: https://unpkg.com/wavesurfer.js@2.2.1/dist/wavesurfer.min.js (You can change the version no. as well) will it be useful?

NishealJ replied:
Yes this builds as expected with other libs in third_party/generated hence this file can be used.
I'll test this and update you asap Thanks

DubeySandeep replied:
@NishealJ, I'm not sure whether [the v2+ will work with our setup](https://wavesurfer-js.org/v2/), if it doesn't work yuo can use 1.4.0:  https://unpkg.com/wavesurfer.js@1.4.0/dist/wavesurfer.min.js

NishealJ replied:
@DubeySandeep I've tried using v2+ as zip previously but failed because v2+ has completely modular code which is difficult to combine with our third_party.js as they were having a call to modules which were not available in /generated.

but the CDN seems working good as it is minified and made for distribution so has no modular code [here](https://drive.google.com/open?id=12TlzOx-XHeBFH89WhXkNY0nAwKojS1Tq) is a quick video of v2+ PTAL Thanks!

DubeySandeep commented:
@NishealJ, I reviewed the "UI test" doc and left a few comments. PTAL!

NishealJ committed:
Update version

NishealJ commented:
> @NishealJ, I reviewed the "UI test" doc and left a few comments. PTAL!

Hi @DubeySandeep, I've addressed all your comments (here and [in docs](https://docs.google.com/document/d/1Jld0h1QGnmjfnELu6dH3V5UXIgCpSOTE-knbv_1dBOw/edit)) and committed the needed changes PTAL I'll update provide you an update on testing of new changes soon Thanks!

DubeySandeep reviewed (approved):
@NishealJ, Changes LGTM!

DubeySandeep commented:
@apb7 & @vojtechjelinek, PTAL!

apb7 reviewed (approved):
LGTM, thanks @NishealJ!
